django/db/models/aggregates.py

```diff
@@ -397,3 +400,128 @@ def __init__(self, expression, sample=False, **extra):
 
     def _get_repr_options(self):
         return {**super()._get_repr_options(), "sample": self.function == "VAR_SAMP"}
+
+
+class JSONArrayAgg(Aggregate):
+    function = "JSON_ARRAYAGG"
+    output_field = JSONField()
+    allow_order_by = True
+    arity = 1
+
+    def __init__(self, *expressions, absent_on_null=False, **extra):
+        self.absent_on_null = absent_on_null
+        super().__init__(*expressions, **extra)
+
+    def as_sql(self, compiler, connection, **extra_context):
+        if self.filter and not connection.features.supports_aggregate_filter_clause:
+            raise NotSupportedError(
+                "JSONArrayAgg(filter) is not supported on this database backend."
+            )
+        if self.absent_on_null and not connection.features.supports_json_absent_on_null:
+            raise NotSupportedError(
+                "JSONArrayAgg(absent_on_null) is not supported on this database "
+                "backend."
+            )
+        return super().as_sql(compiler, connection, **extra_context)
+
+    def as_mysql(self, compiler, connection, **extra_context):
+        if self.order_by is not None:
+            raise NotSupportedError(
+                "JSONArrayAgg(order_by) is not supported on this database backend."
+            )
+        return self.as_sql(compiler, connection, **extra_context)
+
+    def as_sqlite(self, compiler, connection, **extra_context):
+        sql, params = self.as_sql(
+            compiler, connection, function="JSON_GROUP_ARRAY", **extra_context
+        )
+        # JSON_GROUP_ARRAY defaults to returning an empty array on an empty
+        # set. Modifies the SQL to support a custom default value to be
+        # returned, if a default argument is not passed, null is returned
+        # instead of [].
+        if (default := self.default) == []:
+            return sql, params
+        # Ensure Count() is against the exact same parameters (filter,
+        # distinct)
+        count = self.copy()
+        count.__class__ = Count
+        count_sql, count_params = compiler.compile(count)
+        default_sql = ""
+        default_params = ()
+        if default is not None:
+            default_sql, default_params = compiler.compile(default)
+            default_sql = f" ELSE {default_sql}"
+        sql = f"(CASE WHEN {count_sql} > 0 THEN {sql}{default_sql} END)"
+        return sql, count_params + params + default_params
+
+    def as_native(self, compiler, connection, *, returning=None, **extra_context):
+        # Oracle and PostgreSQL 16+ default to removing SQL null values from
+        # the returned array. This adds the NULL ON NULL clause to preserve
+        # the null values in the array as default behaviour Similar to that
```

Comment:
Minor readability nit: there’s a small punctuation issue in this comment.
```suggestion
        # the null values in the array as default behaviour. Similar to that
```